report_builder/utils.py

```diff
@@ -31,3 +34,65 @@ def duplicate(obj, changes=None):
             except:
                 pass
     return duplicate
+
+
+DATE = 1
+NUMBER = 2
+
+
+def sort_helper(x, sort_key, sort_type):
+    """ Sadly python 3 makes it very hard to sort mixed types
+        We can work around this by forcing the types
+    """
+    result = x[sort_key]
+    if result is None:
+        if sort_type == DATE:
+            result = datetime.date(datetime.MINYEAR, 1, 1)
+        elif sort_type == NUMBER:
+            result = 0
+        else:  # Last try - make it a string
+            result = ''
+    return result
+
+
+def sort_data(data_list, display_field):
+    position = display_field.position
+    is_reverse = display_field.sort_reverse
+    # Try to inspect sample data to determine type
+    sample_data = data_list[0][position]
+    if sample_data is None:
+        sample_data = data_list[-1][position]
+    sort_type = None
+    if isinstance(sample_data, (datetime.date, datetime.datetime)):
+        sort_type = DATE
+    elif isinstance(sample_data, (int, float, complex)):
+        sort_type = NUMBER
+    return sorted(
+        data_list,
+        key=lambda x: sort_helper(x, position, sort_type),
+        reverse=is_reverse
+    )
+
+
+def increment_total(display_field, data_row):
```

Comment:
maybe it would make sense to move this to a [static method](http://stackoverflow.com/a/735978/907060) of a model. it seems linked very closely with model data and isnt a very generic gunction